### PR TITLE
Fix various crashes

### PR DESCRIPTION
**Pull request by bmarty**
## Content

Fix various crashes reported by Sentry.
Also cleanup the code at a few location.

To be reviewed commit per commit.

## Motivation and context

More stable application.

## Tests

Not always easy to repro the crashes.
For the PDF crash, changing the extension of a Gif file (for instance), then send it to a room and try to open it made the application crash. It's no fixed and display an inlined error. 

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: libraries/mediapickers/api/src/main/kotlin/io/element/android/libraries/mediapickers/api/PickerLauncher.kt [hunk 1]
```diff
@@ -25,11 +27,19 @@ class ComposePickerLauncher<Input, Output>(
     private val defaultRequest: Input,
 ) : PickerLauncher<Input, Output> {
     override fun launch() {
-        managedLauncher.launch(defaultRequest)
+        try {
+            managedLauncher.launch(defaultRequest)
+        } catch (activityNotFoundException: ActivityNotFoundException) {
+            Timber.w(activityNotFoundException, "No activity found")
```
Comment (jmartinesp) [inline libraries/mediapickers/api/src/main/kotlin/io/element/android/libraries/mediapickers/api/PickerLauncher.kt:33]: Should we use `toast` int these 2 cases too besides just logging the issue?

File: libraries/mediaviewer/api/src/main/kotlin/io/element/android/libraries/mediaviewer/api/local/pdf/PdfRendererManager.kt [hunk 2]
```diff
@@ -25,20 +28,28 @@ class PdfRendererManager(
...
-                        val firstPages = loadPages(from = 0, to = 3)
-                        mutablePdfPages.value = firstPages
-                        val nextPages = loadPages(from = 3, to = pageCount)
-                        mutablePdfPages.value = firstPages + nextPages
-                    }
+                    runCatching {
+                        PdfRenderer(parcelFileDescriptor)
+                    }.fold(
+                        onSuccess = { pdfRenderer ->
+                            this@PdfRendererManager.pdfRenderer = pdfRenderer
```
Comment (jmartinesp) [inline libraries/mediaviewer/api/src/main/kotlin/io/element/android/libraries/mediaviewer/api/local/pdf/PdfRendererManager.kt:42]: Maybe directly do:

```kotlin
runCatching {
    pdfRenderer = PdfRenderer(parcelFileDescriptor)
}
```

So we get rid of the ugly `this@pdfRendererManager.pdfRenderer = ... ` line.

**Review by jmartinesp — APPROVED** [review]
Thanks, I have a couple of suggestions.

File: libraries/mediapickers/api/src/main/kotlin/io/element/android/libraries/mediapickers/api/PickerLauncher.kt (hunk 1 above)
Comment (bmarty) [reply libraries/mediapickers/api/src/main/kotlin/io/element/android/libraries/mediapickers/api/PickerLauncher.kt:33]: The issue happen when the device has no Camera application, so I would not bother too much.

File: libraries/mediaviewer/api/src/main/kotlin/io/element/android/libraries/mediaviewer/api/local/pdf/PdfRendererManager.kt (hunk 2 above)
Comment (bmarty) [reply libraries/mediaviewer/api/src/main/kotlin/io/element/android/libraries/mediaviewer/api/local/pdf/PdfRendererManager.kt:42]: ugly yes, but I need the `runCatching` to return the value.

Maybe using a also block like

```kotlin
runCatching {
    PdfRenderer(parcelFileDescriptor).also {
        pdfRenderer = it
    }
}.fold(
    onSuccess = { pdfRenderer ->
        // Preload just 3 pages so we can render faster
        val firstPages = pdfRenderer.loadPages(from = 0, to = 3)
        mutablePdfPages.value = AsyncData.Success(firstPages.toImmutableList())
        val nextPages = pdfRenderer.loadPages(from = 3, to = pdfRenderer.pageCount)
        mutablePdfPages.value = AsyncData.Success((firstPages + nextPages).toImmutableList())
    },
    onFailure = {
        mutablePdfPages.value = AsyncData.Failure(it)
    }
)
```

is less ugly?

Comment (bmarty) [reply libraries/mediaviewer/api/src/main/kotlin/io/element/android/libraries/mediaviewer/api/local/pdf/PdfRendererManager.kt]: I have found another solution 55704f02273778214dfe8d184647b12e8d8440de

Comment (jmartinesp) [reply libraries/mediaviewer/api/src/main/kotlin/io/element/android/libraries/mediaviewer/api/local/pdf/PdfRendererManager.kt]: I think the change is ok and maybe I'm overlooking something but with the change I proposed wouldn't the `pdfRenderer` property only be assigned if the initiailzation using the `parcelFileDescriptor` works too?

**Review by ganfra — APPROVED** [review]

File: libraries/mediaviewer/api/src/main/kotlin/io/element/android/libraries/mediaviewer/api/local/pdf/PdfRendererManager.kt (hunk 2 above)
Comment (bmarty) [reply libraries/mediaviewer/api/src/main/kotlin/io/element/android/libraries/mediaviewer/api/local/pdf/PdfRendererManager.kt]: Yes, but with the change you proposed, the result of the assignation is Unit, so I can't get the `PdfRenderer` instance in the `onSuccess` parameter block.

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/room/RustMatrixRoom.kt [hunk 3]
```diff
@@ -105,8 +105,12 @@ class RustMatrixRoom(
 
     override val roomInfoFlow: Flow<MatrixRoomInfo> = mxCallbackFlow {
         launch {
-            val initial = innerRoom.roomInfo().let(matrixRoomInfoMapper::map)
-            channel.trySend(initial)
+            runCatching { innerRoom.roomInfo() }
+                .getOrNull()
+                ?.let(matrixRoomInfoMapper::map)
+                ?.let { initial ->
+                    channel.trySend(initial)
```
[the remark this answers is not in the document]
Comment (bmarty) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/room/RustMatrixRoom.kt:112]: I did not want to change the existing code here. Maybe in another PR?

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/room/RustMatrixRoom.kt [hunk 4]
```diff
@@ -105,8 +105,12 @@ class RustMatrixRoom(
 
     override val roomInfoFlow: Flow<MatrixRoomInfo> = mxCallbackFlow {
         launch {
```
[the remark this answers is not in the document]
Comment (bmarty) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/room/RustMatrixRoom.kt:107]: Yes, will try. I did not want to change the logic.

[the remark this answers is not in the document]
Comment (bmarty) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/room/RustMatrixRoom.kt]: 53fc2f3dfeb8ed0f04b7cb070478b4d31aa6513b

**Pull request merged**